pkg/components/internal/testutil/jsonpath.go

```diff
@@ -102,3 +102,16 @@ func MatchJSONPathInt64Value(t *testing.T, yamlConfig string, jsonPath string, e
 		t.Fatalf("Expected: %d, Got: %d", expected, got)
 	}
 }
+
+// JSONPathExists checks if the given YAML config has an object at the given JSON path, also provide
+// what error to expect.
+func JSONPathExists(t *testing.T, yamlConfig string, jsonPath string, errExp string) {
+	_, err := jsonPathValue(yamlConfig, jsonPath)
+	if err != nil && errExp == "" { //nolint:gocritic
```

Comment:
I would avoid disabling gocritic.

Reply:
It is suggesting usage of a switch case instead of the if else. Given the kinda checks we have switch case is not ideal.

Reply:
It also uses `else`s a lot, which are discouraged in Go..

Reply:
I think we can just remove the else's here.

Reply:
We could also use a const error here, then complexity of the whole thing is reduced to just `errors.Is()` call, which I think is elegant. I can share the code for that.

Reply:
Removed else because they are each func is just a way to exit out of the function.

Reply:
How is `errors.Is()` or any const we create going to help with the error coming from upstream lib? Especially when they have not declared an error var for us to consume but just an error created using `fmt.Errorf`.

https://github.com/kinvolk/lokomotive/blob/8e39195bd021d3ad601268b17f70d53785affe58/vendor/k8s.io/client-go/util/jsonpath/jsonpath.go#L428

Reply:
>How is errors.Is() or any const we create going to help with the error coming from upstream lib? Especially when they have not declared an error var for us to consume but just an error created using fmt.Errorf.

If we set `allowMissingKeys` for jsonpath, then we get no value, so we can return constant `ValueNotFound` error.

Though I'm not a big fan of this const error pattern, as it adds bunch of boilerplate. It would look something like:
```go
type NotFoundError string

func (e NotFoundError) Error() string {
  return string(e)
}

const ErrNotFound NotFoundError = "not found"

type KeyNotFoundError struct {
  Key string
}

func (e KeyNotFoundError) Error() string {
  return fmt.Sprintf("%s %s", e.Key, ErrNotFound)
}

func (e KeyNotFoundError) Unwrap() error {
  return ErrNotFound
}

func jsonPathValue(data, key string) (string, error) {
  if !strings.Contains(data, key) {
    return "", KeyNotFoundError{Key: key}
  }

  return "", nil
}

func JSONPathExistsSimplified(t *testing.T, yamlConfig string, jsonPath string, errExp error) {
  _, err := jsonPathValue(yamlConfig, jsonPath)
  if !errors.Is(err, errExp) {
    t.Fatalf("Unexpected error: %v, expected: %v", err, errExp)
  }
}
```

So I think returning `nil` value would probably also work fine.